### PR TITLE
Change default gene panel for MAF cases from None to OMIM-AUTO

### PR DESCRIPTION
**Pull request by peterpru**
## Description
With recent changes, when the attribute 'gene panel' is empty for a case in MIP-DNA, with delivery type ANALYSIS_FILES such as for MAF controls, it causes the delivery step of that case to fail.

There are two options to fix this problem. As the MAF cases are run through the mip-dna pipeline to generate a VCF file, the most straightforward fix is to add the OMIM gene panel so that the delivery report is completed. The alternative would be to create a new pipeline just for MAF cases consisting of a mapper and a variant caller. The current suggestion here is to complete the delivery, as we have run the full mip pipeline for which delivery is the natural conclusion.

### Added

-

### Changed

- Change default gene panel for MAF cases from None to OMIM-AUTO

### Fixed

-

### How to prepare for test
- [ ] In github: Action > Deploy branch to staging environment > empty-gene-panel-field.

### How to test
- [ ] In https://clinical-stage.scilifelab.se/ make an order for a fastq delivery.

### Expected test outcome
- [ ] check that the automatically created MAF case has as gene panel "OMIM-AUTO"
- [ ] Take a screenshot and attach or copy/paste the output.
image

## Review
- [ ] code approved by
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/orders/fastq_submitter.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def create_maf_case(self, sample_obj: models.Sample):
             data_analysis=Pipeline(Pipeline.MIP_DNA),
             data_delivery=DataDelivery(DataDelivery.ANALYSIS_FILES),
             name="_".join([sample_obj.name, "MAF"]),
-            panels=None,
+            panels=["OMIM-AUTO"],
```
Comment (henrikstranneheim) [inline cg/meta/orders/fastq_submitter.py:64]: I think this is called `OMIM-AUTO`. Should we add a constant class in: cg/constants/gene_panel.py that defines the panel ids? And also make the `MASTER_LIST` use that class

Comment (peterpru) [reply cg/meta/orders/fastq_submitter.py]: I have now changed MasterList to a constant class, and refactored it everywhere I could see it be used.

File: cg/constants/gene_panel.py [hunk 2]
```diff
@@ -1,30 +1,32 @@
...
+    MTDNA: str = "mtDNA"
+    NBS_M: str = "NBS-M"
+    NEURODEG: str = "NEURODEG"
+    NMD: str = "NMD"
+    PEDHEP: str = "PEDHEP"
+    PID: str = "PID"
+    PIDCAD: str = "PIDCAD"
+    OMIM_AUTO: str = "OMIM-AUTO"
+    SKD: str = "SKD"
+
```
Comment (henrikstranneheim) [inline cg/constants/gene_panel.py:32]: How about adding a classmethod here:

```
@classmethod
    def get_panel_names(cls, MasterList: MasterList) -> List[str]:
        """Return pnale names from Master list."""
        return [panel_name.value for panel_name in MasterList]
```

File: cg/meta/workflow/mip.py [hunk 3]
```diff
@@ -159,8 +159,8 @@ def write_panel(self, case_id: str, content: List[str]):
     def convert_panels(customer: str, default_panels: List[str]) -> List[str]:
         """Convert between default panels and all panels included in gene list."""
         # check if all default panels are part of master list
-        if customer in COLLABORATORS and all(panel in MASTER_LIST for panel in default_panels):
-            return MASTER_LIST
+        if customer in COLLABORATORS and all(panel in dir(MasterList) for panel in default_panels):
```
Comment (henrikstranneheim) [inline cg/meta/workflow/mip.py:162]: And using the classmethod here

**Review by henrikstranneheim — COMMENTED** [review]
👍

**Comment by henrikstranneheim** [comment]
@peterpru Should we also take the opportunity to rename MasterList to something more intuitive? The name should include "gene_panel" or similar at least to reflect what it is a master list of.

File: cg/meta/workflow/mip.py (hunk 3 above)
Comment (peterpru) [reply cg/meta/workflow/mip.py]: Added.

File: cg/constants/gene_panel.py (hunk 2 above)
Comment (peterpru) [reply cg/constants/gene_panel.py]: Great, I have now added the classmethod.

**Comment by peterpru** [comment]
@henrikstranneheim changing the name sounds good, I have used GenePanelMasterList, which feels a bit long, but does reflect its contents.

File: cg/constants/gene_panel.py [hunk 4]
```diff
@@ -1,30 +1,40 @@
...
+    @classmethod
+    def get_panel_names(cls, panels=None) -> List[str]:
+        """Return requested panel names from Master list, or all panels if none are specified."""
+        if not panels:
+            return list(map(lambda p: p.value, cls))
```
Comment (henrikstranneheim) [inline cg/constants/gene_panel.py:39]:
```suggestion
            return list(map(lambda panel: panel.value, cls))
```
Avoid single character variables if possible they are hard to understand and not mnemonic

**Review by henrikstranneheim — APPROVED** [review]
Well done!

File: cg/constants/gene_panel.py (hunk 4 above)
Comment (peterpru) [reply cg/constants/gene_panel.py:39]: That sounds good.

**Comment by peterpru** [comment]
Deploying in production:
image

**Pull request merged**